gobblin-audit/src/main/java/gobblin/audit/values/sink/FsAuditSink.java

```diff
@@ -0,0 +1,120 @@
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.audit.values.sink;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URI;
+
+import lombok.Getter;
+
+import org.apache.avro.file.DataFileWriter;
+import org.apache.avro.generic.GenericDatumWriter;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import com.google.common.io.Closer;
+import com.typesafe.config.Config;
+
+import gobblin.annotation.Alias;
+import gobblin.audit.values.auditor.ValueAuditGenerator;
+import gobblin.audit.values.auditor.ValueAuditRuntimeMetadata;
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.util.ConfigUtils;
+import gobblin.util.PathUtils;
+
+
+/**
+ *  A Hadoop {@link FileSystem} based {@link AuditSink} that writes audit {@link GenericRecord}s to a file on {@link FileSystem}.
+ *  <ul>
+ *  <li> The {@link FileSystem} {@link URI} can be set using key {@link ConfigurationKeys#FS_URI_KEY}, {@link ConfigurationKeys#LOCAL_FS_URI} is used by default.
+ *  <li> All audit files are written under the base path. The base path can be set using key {@link #FS_SINK_AUDIT_OUTPUT_PATH_KEY}.
+ *  The default path is,
+ *  <pre>
+ *  <code>System.getProperty("user.dir") + "/lumos_value_audit/local_audit";</code>
+ *  </pre>
+ *  <li> It uses <code>auditMetadata</code> to build the audit file name and path.<br>
+ *  <b>The layout on {@link FileSystem} - </b>
+ *  <pre>
+ *  |-- &lt;Database&gt;
+ *    |-- &lt;Table&gt;
+ *       |-- P=&lt;PHASE&gt;.C=&lt;CLUSTER&gt;.E=&lt;EXTRACT_ID&gt;.S=&lt;SNAPSHOT_ID&gt;.D=&lt;DELTA_ID&gt;
+ *          |-- *.avro
+ *  </pre>
+ *  </ul>
+ */
+@Alias(value = "FsAuditSink")
+public class FsAuditSink implements AuditSink {
+
+  private static final String FS_SINK_AUDIT_OUTPUT_PATH_KEY = ValueAuditGenerator.VALUE_AUDIT_CONFIG_KEY_PREFIX + "sink.fs.output.path";
+  private static final String FS_SINK_AUDIT_OUTPUT_DEFAULT_PATH = System.getProperty("user.dir") + "/lumos_value_audit/local_audit";
+  private static final String FILE_NAME_DELIMITTER = "_";
+
+  private final FileSystem fs;
+  private final OutputStream auditFileOutputStream;
+  private final DataFileWriter<GenericRecord> writer;
+  private final Closer closer = Closer.create();
+  private final ValueAuditRuntimeMetadata auditMetadata;
+  @Getter
+  private final Path auditDirPath;
+
+  public FsAuditSink(Config config, ValueAuditRuntimeMetadata auditMetadata) throws IOException {
+
+    this.fs = FileSystem.get(URI.create(ConfigUtils.getString(config, ConfigurationKeys.FS_URI_KEY, ConfigurationKeys.LOCAL_FS_URI)), new Configuration());
```

Comment:
I suggest that we allow the path to have a file-system part instead of relying on yet another property. The only downside is that we can't have relative paths for the local file system but I think that should be OK.